### ISSUE TITLE
What does "error retrieving device certificate: CMS verify MDM Signed Message" mean, seems something expired?

### ISSUE THREAD

**Issue by GregOriol** [micromdm/micromdm#1009]
### What version of `micromdm` are you using?
v1.21.1

### What `micromdm` command did you run?
micromdm serve

micromdm has been set up more than a year ago
trying to issue some commands for a device that has been enrolled some time ago (maybe more than a year?)

### What did you expect to see?/What did you see instead?
The device used here did work/receive/perform commands before, but now I get this message in the logs:

```
err="error retrieving device certificate: CMS verify MDM Signed Message: pkcs7: signing time \"2024-06-13T07:00:24Z\" is outside of certificate validity \"2023-06-06T13:07:58Z\" to \"2024-06-05T13:07:58Z\""
transport=http method=PUT status=500 proto=HTTP/1.0 host=127.0.0.1 user_agent=MDM/1.0 path="/mdm/connect"
```

We did upgrade certs in April/May as described in https://github.com/micromdm/micromdm/blob/main/docs/user-guide/quickstart.md#renewing-certificates and devices did performs commands as expected after that.

Now I guess something else has expired, but what exactly and how to renew? (the validity period in the message seems to be different from one device to another)

**Comment by jessepeterson** [comment]
Yes, looks like your device's enrollment expired some days ago. MicroMDM has a default 1 year expiry on the enrollment profile [identity certificate](https://micromdm.io/blog/certificates/).

Some additional information here for macOS: https://github.com/micromdm/micromdm/wiki/Device-Identity-Certificate-Expiration :

> By default MicroMDM's SCEP server issues certificates with a 1-year expiry.

**Comment by GregOriol** [comment]
@jessepeterson Oh, I did look around the repo but didn't find this page... Could be it! I'll check

**Comment by GregOriol** [comment]
Indeed, reinstalling an enrollment profile fixes the issue.

**Comment by jessepeterson** [comment]
@GregOriol you may want to consider raising the validity duration with the `-scep-client-validity` switch in the MicroMDM server. As well you can send enrollment profiles _over the MDM channel_ to have a device re-enroll itself to renew its certificate via MDM.

**Comment by GregOriol** [comment]
@jessepeterson Yes we implemented an automatic new profile push, plus manual installation for those having expired. However we decided not to change the validity period as we didn't know what it impacts. Would it be ok to set 10000 days for example?

**Comment by pawandatacultr** [comment]
> Indeed, reinstalling an enrollment profile fixes the issue.

@GregOriol How do we reinstall enrolment profile?

**Comment by korylprince** [comment]
@pawandatacultr you can install the enrollment profile like any other profile, with the InstallProfile command. This will cause the device to renew its enrollment.
